programs/mango-v4/src/state/group.rs

```diff
@@ -69,11 +71,75 @@ impl Group {
         self.is_testing() || self.version > 1
     }
 
-    pub fn is_operational(&self) -> bool {
-        self.halted == 0
+    pub fn is_ix_enabled(&self, ix: IxGate) -> bool {
+        self.ix_gate & (1 << ix as u128) == 0
     }
 }
 
+/// Enum for lookup into ix gate
+/// note:
+/// total ix files 56,
+/// ix files included 47,
+/// ix files not included 9,
+/// - Benchmark,
+/// - ComputeAccountData,
+/// - GroupCreate
+/// - GroupEdit
+/// - HealthRegion
+/// - IxGateSet,
+/// - PerpZeroOut,
+/// - PerpEditMarket,
+/// - TokenEdit,
+pub enum IxGate {
+    AccountClose = 0,
+    AccountCreate,
```

Comment:
I'd even suggest explicitly numbering all enum variants, so it's impossible to accidentally reuse/reshuffle numbers